core/container.go

```diff
@@ -226,7 +230,19 @@ func (container *Container) Exec(ctx context.Context, gw bkgw.Client, args []str
 		llb.AddMount(shim.Path, shimSt, llb.SourcePath(shim.Path)),
 		llb.Args(append([]string{shim.Path}, args...)),
 		llb.WithCustomName(strings.Join(args, " ")),
-		llb.AddMount(metaMount, llb.Scratch()),
+
+		// create /dagger mount point for the shim to write to
+		llb.AddMount(
+			metaMount,
+			// because the shim might run as non-root, we need to make a
```

Comment:
Just a note: I'm actually not sure whether we'd really want the shim to run as the same user in the long run. Some of the ideas we have for it in the future involve networking stuff that might require root in the network namespace.

This is totally fine for now since updating the shim to do setuid is a lot more work and this implementation is not breaking anything now. Just wanted to bring it up for future thought. cc @aluzzardi